lib/Backend/GlobOpt.cpp

```diff
@@ -8909,21 +8910,30 @@ GlobOpt::TypeSpecialization(
             {
                 return instr;
             }
+
+            // need to run typespec if cannot const fold InlineMathAbs, or dst will be dead
+            if (instr->m_opcode != Js::OpCode::InlineMathAbs)
+            {
+                runTypeSpec = false;
```

Comment:
Yes, bailout handles run-time behavior, but the error here is compile time assertion. LoopPre pass does type specialize on dest of Math.abs(MIN_INT) to int32, and convert this dest to float when folding. Assertion is triggered when merging current block to successor loop header block which restores states on back edge, because it cannot convert float to lossless int32.